barretenberg/cpp/src/barretenberg/translator_vm/translator_proving_key.hpp

```diff
@@ -3,6 +3,7 @@
 
 #include "barretenberg/translator_vm/translator_flavor.hpp"
 namespace bb {
+// WORKTODO: This class likely should not exist. Any reason not to use the ProvingKey in the flavor?
```

Comment:
yes, I created this class to unify where translator-related polynomials are being constructed, previously code was spread between the prover and the flavor in a quite arbitrary way that was making it hard to follow. I am of the opinion that the `ProvingKey` in the flavor should go away (same like the case for `DeciderProvingKey` and `ProvingKey`). My morning brain is thinking that perhaps `ProvingKey` should be a base class that `TranslatorProvingKey` / `ECCVMProvingKey` (there things are still way messier) / `DeciderProvingKey` derive